moped-editor/src/views/projects/projectView/ProjectSummary/ProjectSummaryKnackDataTrackerSync.js

```diff
@@ -100,27 +146,55 @@ const ProjectSummaryKnackDataTrackerSync = ({
         .then(result => {
           if (result.errors) {
             // Successful HTTP request, but knack indicates an error with the query, such as non-existent ID.
-            // Reject the promise to fall through to the .catch() method
-            return Promise.reject(result);
+            // Throw error & cause the promise to fall through to the .catch() method
+            throw result;
           } else {
             // Successful HTTP request with meaningful results from Knack
             project.currentKnackState = result; // this assignment operates on `project` which is defined in broader scope than this function
-            return fetch(knackEndpointUrl, {
-              // fetch returns a Promise for the next step
-              method: knackHttpMethod,
-              headers: buildHeaders,
-              body: buildBody(),
-            });
+
+            return countSignalsInProject(project)
+              .then(signalCount => {
+                if (signalCount > 0) {
+                  return fetch(
+                    knackSignalEndpointUrl +
+                      "?filters=" +
+                      buildSignalIdFilters(project),
+                    {
+                      // Fetch will return a promise, which we'll use to start a chain of .then() steps
+                      method: "GET",
+                      headers: buildHeaders,
+                    }
+                  )
+                    .then(response => response.json()) // get the json payload and pass it along
+                    .then(result => {
+                      let signalIds = [];
+                      if (signalCount > 0) {
+                        signalIds = result.records.map(record => record.id);
+                      }
+                      return signalIds;
+                    });
+                } else {
+                  return [];
+                }
+              })
+              .then(signalIds => {
+                console.log("about to act with: " + knackHttpMethod);
+                return fetch(knackProjectEndpointUrl, {
+                  method: knackHttpMethod,
+                  headers: buildHeaders,
+                  body: buildBody(signalIds),
+                });
+              });
           }
         })
         .then(response => response.json())
         .then(result => {
           if (result.errors) {
             // Successful HTTP request, but knack indicates an error with the query, such as non-existent ID
-            return Promise.reject(result);
+            throw result;
           } else {
             // Successful HTTP Update request with meaningful results from Knack
-            return Promise.resolve();
+            return true;
```

Comment:
Is this right? What is the least more true than false?

Reply:
this is like a galaxy brain question for me this morning

Reply:
if you're not doing anything useful later with `true` (you're just continuing the chain), then i think you can just nix the `else {}`  block entirely and let the function return `undefined`. imo that would be better, because i otherwise read this to mean that there's a dependency on the return value `true`.

Reply:
@johnclary - thank you! I couldn't see the forest for the trees there. That else block used to have some wayward functionality that slowly got chipped away until it was an empty non-op.